userStoryInventoryChangHF.md

```diff
@@ -39,41 +39,646 @@ This section shows the areas or sources thus far
+
+### Acceptance Criteria
+
+- Given that I am an Ada holder who is logged into the app and has delegated their voting power to a dRep.
+- When I use the delegate to dRep page.
+- Then I can use it to delegate all my voting power to a different dRep.
+
+## 2G: Retire as dRep
+
+**User Story:** As a dRep, I want to be able to retire so that I can stop being a dRep.
+
+### Acceptance Criteria
+
+- Given that I am logged into the app.
+- When I press the retire as dRep button.
+- Then my wallet asks me to sign the transaction to retire from being a dRep.
+
+## 2H: State After Retiring as dRep
+
+**User Story:** As a retired dRep, I want to be able to confirm that I am no longer registered as a dRep so that I have peace of mind.
+
+### Acceptance Criteria
+
+- Given that I am logged into the app with the wallet account I used when I registered as a dRep.
+- When I am on the governance action page.
+- Then I no longer have the option to vote. Furthermore, on all pages, there is no widget informing me of my dRepID or about how much voting power has been delegated to me.
+
+## 2I: Check Valid dRep ID
+
+**User Story:** As a user of the app, I want to ensure that I am delegating to a valid dRep ID so that I do not make a mistake.
+
+### Acceptance Criteria
+
+- Given I have my wallet connected to the app.
+- When I attempt to delegate to a dRep using their dRep ID (such as described in 2D & 2H). I press the button to submit my delegation to the wallet.
+- Then before the app passes my data to the wallet, it will check it against a list of all registered dRep IDs. If the dRep ID doesn’t exist, It will give an error message to the user and not allow them to delegate to the invalid dRep ID.
+
+## 2J: Prevent Delegation with Insufficient Funds
+
+**User Story:** As either an Ada Holder or DRep, I want to be prevented from delegating when I have insufficient funds.
+
+### Acceptance Criteria
+
+- Given that I have connected to GovTool with zero ADA (or tADA in the case of SanchoNet).
+- When I choose the Delegate to DRep ID option and enter a registered DRep ID and press the Delegate button.
+- Then I am presented with a warning message and cannot proceed with delegation if my balance is zero or below the transaction costs.
+
+## 2K: Delegation page should be accessible on disconnected state
+
+**User Story:** As a user without a compatible wallet connected to GovTool, I want the delegation page to be accessible.
+
+### Acceptance Criteria
+
+- Given that I do not have a compatible wallet connected to GovTool.
+- When I attempt to visit the URL of the Delegate to DRep page.
+- Then I can see the delegation options.
+
+## 2L: Delegate to Self for Registered DReps
+
+**User Story:** As a registered DRep connected to GovTool with my wallet, I want to be able to delegate to myself via my connected wallet on the Delegate to DRep page.
```

Comment:
Currently on GovTool, when a user registers to be a DRep, GovTool forces them to delegate to themself in the same transaction